src/main/drivers/pwm_output_dshot_hal.c

```diff
@@ -61,6 +62,12 @@ FAST_CODE void pwmWriteDshotInt(uint8_t index, uint16_t value)
         return;
     }
 
+    /*If there is a command ready to go overwrite the value and send that instead*/
+    if (pwmIsDshotCommand()) {
```

Comment:
What say you, @DieHertz? Done this way, this puts a pretty good limit on the amount of additional code to go into FAST_RAM.

Should `pwmIsDshotCommand()` and `pwmProcessingDshotCommand()` be explicitly inlined, to ensure that they are always put into FAST_RAM?

Reply:
You mean into ITCM-RAM (FAST_CODE storage specifier? Fastram is non-executable :)

I need to read code in this PR again to get the idea

Reply:
I don't think I understand.
Why you're suggesting to inline these functions, if their code was extracted to avoid ITCM-RAM overflow?

Reply:
Sorry, now that I am looking at this in more detail I see that this is wrong:
- `pwmIsDshotCommand()` should probably be `pwmIsProcessingDshotCommand()` to make it more clear, and should be inlined because it's in the critical path for Dshot motor updates;
- `pwmProcessingDshotCommand()` should be `pwmProcessDshotCommand()` and not be inlined;
- `pwmProcessDshotCommand()` should be guarded by `pwmIsProcessingDshotCommand();

Reply:
Explicit inline is not necessary, compiler will decide on it's in with our `-O` levels anyway.
What needs to be done is mark the function that does the main job as `NOILINE`

Reply:
Agreed re `NOINLINE`. Is the compiler clever enough to spot that `pwmIsProcessingDshotCommand()` should always be inlined if it's called from  ITCM-RAM? (Another, probably better strategy would be to declare it `FAST_CODE`.)

Reply:
It's hard to find a case where the compiler decided not to inline a function in our codebase :)
Marking `FAST_CODE` is ok for stating the intention and making it clear-er for new/future devs.

Reply:
Yeah, I assumed that much. The aspect of making users aware that this has to fit into the limited `FAST_CODE` space by explicitly marking it so is a good point - we can't expect them to go and check all callers for this before modifying a function.

Reply:
OK I made these changes.